tokens_definition_example_draft_v2.json

```diff
@@ -0,0 +1,205 @@
+{
+	"tokenCategories" : [
+		{
+			"label": "general",
+			"name": "general"
+		},
+		{
+			"label": "layout",
+			"name": "layout"
+		}
+	],
+	"tokenSets" : [
+		{
+			"label": "grid",
+			"name": "grid"
+		},
+		{
+			"label": "links",
+			"name": "links"
+		}
+	],
+	"tokens": [
+		{
+			"mappings" : [
+				{
+					"name": "CSSVariable",
```

Comment:
Hey @ruben-pulido, sorry, I think there was a confusion here on what we were asking to map. You're mapping "abstract" values to "implementation" values, which is interesting to represent those but won't help in implementing this.

I think that what we need is something like:

```
"tokens": [
    {
        "name": "linkColor",
        "editorType": "ColorPicker",
        "mappings": [
            {
                "type": "CSSVariable",
                "variable" "link-color"
            }
        ]
    }
]
```

I think @marcoscv-work or @eduardoallegrini can start working based on this document to provide better insight as to how this mapping needs to be done before we dig into this deeper.

Reply:
In that case it is not clear for how the mapping will be done. Looking forward to hear more details on this.

Reply:
Related to this, I'm bringing some comments from the old PR ([more context here](https://github.com/jbalsas/liferay-portal/pull/2218#discussion_r452101451)):

---
I just want to split tokens and CSS variables clearly. If we are going to have two terms it's because they are two different things and, in that case, we need clear demarcation criteria. Otherwise, why are we calling this tokens but then saying everywhere that they are just CSS variables? :shrug: 

If they are CSS variables just because we only have one implementation now, but may have others in the future, it's OK to call them tokens. But let's call them tokens and have a clear image in mind of what else they can be so as not to make a leaky abstraction and develop something that we call tokens but can only be implemented with CSS variables or else it breaks.

----

In this specific case: we are defining an abstract token, which can be made real as a CSS variable, but because tokens are not the same as CSS variables, could be made real as an Android style, GTK style, or _whatever_ we may invent...

If that `whatever` is 70 more things, will we have 70 different mappings in this file called cssVariable, gtkStyle, androidStyle, whatever* to map the token to different things?

----

Is that what we want? 

We may want it, I'm not arguing against it, but in that case, the one defining the tokens (abstract layers) must define how they can be mapped to different things (implementation layer).

Or else, it can also be that I haven't understood the difference between CSS variables and tokens yet. This wouldn't surprise me :roll_eyes:

Reply:
🤔 I'm sorry but I will need some clarification. After reading the [style-editor-documentation](https://docs.google.com/document/d/1Tn8ycrPZQkEtzKNZPF5k7PCeMR6Pkc0Ke7RYW-yiqKI/edit#), [the previous PR](https://github.com/jbalsas/liferay-portal/pull/2218) (@jbalsas you mentioned someone else here https://github.com/jbalsas/liferay-portal/pull/2218#issuecomment-652890351 🤣) and all the comments, I've got a general idea but since I'm not sure I prefer to ask "what exactly I need to define?"

Reply:
[Provide TokenDefinition for Classic Theme](https://issues.liferay.com/browse/LPS-116194).

This should be the task that we need to do. We would need to map the defined "Design Tokens" with "CSSVariables" which values we can control to make those things change.

This `json` schema defines how that mapping and the tokens are defined... but agreeing on the mappings portion is hard, so we could try the other way around, providing a `theme-token-definition.json` file that matches https://www.notion.so/Stylebook-Variables-a919ebecc01046d5bb8f89e076842503

Reply:
We can expand on this in the team channel @eduardoallegrini 😉

Reply:
We've agreed (I think) to try this out with a real example and work our way back to `mappings` from there.